runtime/src/main/scala/com/soundcloud/twinagle/ProtoService.scala

```diff
@@ -0,0 +1,25 @@
+package com.soundcloud.twinagle
+
+import com.twitter.finagle.Service
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.Future
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion}
+
+case class ProtoService(rpcs: Seq[ProtoRpc]) {
+  assert(rpcs.map(_.metadata.service).toSet.size == 1, "inconsistent services in metadata")
+  assert(rpcs.map(_.metadata.prefix).toSet.size == 1, "inconsistent prefixes in metadata")
```

Comment:
(note to self: in-person follow up so I fully understand the possible circumstances here)